pynxtools_xps/scienta/scienta_txt.py

```diff
@@ -35,11 +34,83 @@
     construct_data_key,
     construct_detector_data_key,
     convert_pascal_to_snake,
+    _re_map_single_key,
+    _re_map_single_value,
+)
+from pynxtools_xps.value_mappers import (
+    convert_energy_type,
+    convert_energy_scan_mode,
+    get_units_for_key,
 )
-from pynxtools_xps.value_mappers import convert_energy_type, convert_energy_scan_mode
 
 from pynxtools_xps.scienta.scienta_txt_data_model import ScientaHeader, ScientaRegion
 
+
+def _extract_energy_units(energy_units: str):
+    """
+    Extract energy units from the strings for energy_units.
+    Binding Energy [eV] -> eV
+
+    """
+    return re.search(r"\[(.*?)\]", energy_units).group(1)
+
+
+def _separate_dimension_scale(scale: str):
+    """
+    Seperate the str of the dimension scale into a numpy array
+
+    Parameters
+    ----------
+    scale : str
+        Str of the form "600 599.5 599 598.5 5".
+
+    Returns
+    -------
+    np.ndarray
+        Dimension scale as a numpy array.
+
+    """
+    return np.array([float(s) for s in scale.split(" ")])
+
+
+def _construct_date_time(date_string: str, time_string: str) -> Optional[str]:
+    """
+    Convert the native time format to the datetime string
+    in the ISO 8601 format: '%Y-%b-%dT%H:%M:%S.%fZ'.
+
+    """
+
+    def _parse_date(date_string: str) -> datetime.datetime:
+        possible_date_formats = ["%Y-%m-%d", "%m/%d/%Y"]
+        for date_fmt in possible_date_formats:
+            try:
+                return datetime.datetime.strptime(date_string, date_fmt)
+            except ValueError:
+                pass
+        raise ValueError("Date format not recognized")
```

Comment:
Consider reraising this in the except clause from the actual `ValueError`. That way the error stack is kept.

Reply:
makes sense, changed